src/main/java/com/homework/demo/controller/FilesController.java

```diff
@@ -0,0 +1,82 @@
+package com.homework.demo.controller;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;
+
+import com.homework.demo.modal.FileInfo;
+import com.homework.demo.message.ResponseMessage;
+import com.homework.demo.service.FilesStorageService;
+
+@Controller
+public class FilesController {
```

Comment:
Create a service class where you'll keep all the business logic.

Reply:
Controllers are for web-layer operations only. For example, tuning the response.